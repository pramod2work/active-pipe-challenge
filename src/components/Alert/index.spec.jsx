import React from 'react'
import { shallow } from 'enzyme'

import { Error } from './index'

describe('Error Component test', () => {
  test('Component Snapshot Testing', () => {
    const component = shallow(<Error />)
    expect(component).toMatchSnapshot()
  })

  test('Callback function to be called', () => {
    const props = {
      retryAction: jest.fn()
    }
    const component = shallow(<Error {...props} />)

    component.find('[data-test-id="retry-action"]').last().simulate('click')
    expect(props.retryAction).toBeCalled()
  })
})