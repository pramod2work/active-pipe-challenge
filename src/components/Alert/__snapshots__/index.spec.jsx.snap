// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component test Component Snapshot Testing 1`] = `
<div
  className="error"
  role="alert"
>
  <div
    className="errorSection"
  >
    <div
      className="twoColumnGrid"
    >
      <span
        aria-label="sorry"
        className="errorIcon"
        role="img"
      >
        ðŸ˜ž
      </span>
      Uh-oh... it looks like some things haven't loaded correctly.
    </div>
    <div
      className="twoColumnGrid"
    >
      <button
        aria-label="retry"
        className="retryAction"
        data-test-id="retry-action"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="retry"
          className="retryIcon"
          src="retry.png"
        />
      </button>
    </div>
  </div>
</div>
`;
